### PR TITLE
focalpoints are floats between -1..1, not int

### PR DESCRIPTION
**Pull request by osmaa**
https://docs.joinmastodon.org/api/guidelines/#focal-points

(another thing caught via a sentry alert)

### CODE REVIEW

**Comment by andrewgodwin** [comment]
Does this not need a database migration?

**Comment by osmaa** [comment]
it does. I wasn't sure what to do with the autogenerated file

**Comment by osmaa** [comment]
adding the test case for this i noticed that attachment.to_ap() wasn't producing the (under-specced) Mastodon focalPoint tuple, so that's now added, too.

What I haven't tested is whether defining a focal point through a client app actually goes through end-to-end. I suppose it should, now..

File: activities/models/post_attachment.py [hunk 1]
```diff
@@ -122,6 +122,11 @@ def to_ap(self):
             "mediaType": self.mimetype,
             "blurhash": self.blurhash,
         }
+        if self.is_image() and self.focal_x and self.focal_y:
+            ap["type"] = "Image"
+            ap["mediaType"] = self.mimetype
```
Comment (andrewgodwin) [inline activities/models/post_attachment.py:127]: This seems redundant?

Comment (osmaa) [reply activities/models/post_attachment.py:127]: mediatype, you're right. for type = "Image", not sure. Mastodon sets and sends focalpoint on type = Image attachments, and it doesn't seem right to try to set that on type = Document attachments instead. This is one of those cases where end-to-end interop testing is not simple - it turns out none of the clients I use even support defining a focal point, so I can't send a test message..

**Review by andrewgodwin — APPROVED** [review]

File: activities/models/post_attachment.py (hunk 1 above)
Comment (andrewgodwin) [reply activities/models/post_attachment.py]: Yeah, it was just the double mediaType assignment I was calling out!

**Comment by andrewgodwin** [comment]
Thanks!

**Pull request merged**